cordova-lib/src/cordova/platform_metadata.js

```diff
@@ -0,0 +1,113 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+
+var path         = require('path'),
+    cordova_util = require('./util'),
+    fs           = require('fs'),
+    Q            = require('q'),
+    shelljs      = require('shelljs'),
+    events       = require('../events');
+
+
+// Retrieves the platforms and their versions from the platforms.json file
+// Returns an array of {platform: platform, version: version} ...
+// ... where version could be '3.4.0', '/path/to/platform' or 'git://...'
+function getVersions(projectRoot) {
+    var platformsDir = path.join(projectRoot, 'platforms');
+    var platformsJsonFile = path.join(platformsDir, 'platforms.json');
+
+    // If the platforms.json file doesn't exist, retrieve versions installed
+    // In this case, we won't be able to know what source the platform came from
+    return Q.nfcall(fs.exists, platformsJsonFile).then(function(exists){
+        if(exists){
+            return getPlatVersionsFromFile(platformsJsonFile);
+        }        
+        return getPlatVersionsFromFileSystem(projectRoot);
+    });
+}
+
+// Returns a promise
+function getPlatVersionsFromFile(platformsJsonFile){
+    var platformData = getJson(platformsJsonFile);
```

Comment:
You may want to require JSON file directly: `var platformData = require(platformsJsonFile);`